### PR TITLE
comma: init at 1.1.0

### LINKED ISSUE #2: Add to Nixpkgs
Is there any way we could get this added to Nixpkgs? Doing so would greatly simplify the installation/setup/upgrade process.

### PR DESCRIPTION
**Pull request by Enzime**
###### Motivation for this change

Fixes https://github.com/nix-community/comma/issues/2

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by Enzime** [comment]
@GrahamcOfBorg build comma

**Review by 0x4A6F — APPROVED** [review]
Built and tested on x86_64-linux.

**Comment by Enzime** [comment]
Just fixed the `maintainers` variable

**Comment by 0x4A6F** [comment]
Dependency `nix-index` doesn't build on `aarch64-darwin`, any @NixOS/darwin-maintainers aware of this?

**Comment by prusnak** [comment]
> Dependency `nix-index` doesn't build on `aarch64-darwin`, any @NixOS/darwin-maintainers aware of this?

nix-index hasn't had a stable release since 2018 - this PR https://github.com/NixOS/nixpkgs/pull/153229 updates nix-index to unstable to support aarch64-darwin build

File: pkgs/tools/package-management/comma/default.nix [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  buildInputs = [ nix-index fzy ];
+
+  installPhase = ''
+    install -Dm755 , -t $out/bin
+    ln -s $out/bin/, $out/bin/comma
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/nix-community/comma";
+    description = "Comma runs software without installing it.";
```
Comment (veprbl) [inline pkgs/tools/package-management/comma/default.nix:46]:
```suggestion
    description = "Comma runs software without installing it";
```

Comment (Enzime) [reply pkgs/tools/package-management/comma/default.nix]: Updated :+1:

File: pkgs/tools/package-management/comma/default.nix [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    sha256 = "sha256-IT7zlcM1Oh4sWeCJ1m4NkteuajPxTnNo1tbitG0eqlg=";
+  };
+
+  patches = [
+    (fetchpatch {
+      name = "support-nix-2.4.patch";
+      url = "https://github.com/nix-community/comma/commit/ca8998f4377afcab5904b29ea88fe329716535c8.patch";
+      sha256 = "sha256-e8ic2swpUNLEQ/uBQx9vD7FEBw6aloZ5mPexcsf9oFw=";
+    })
+  ];
```
Comment (Artturin) [inline pkgs/tools/package-management/comma/default.nix:27]: not needed anymore as i merged PRs in the comma repo. i also tagged a new release

File: pkgs/tools/package-management/comma/default.nix [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+  postPatch = ''
+    substituteInPlace , \
+      --replace '--db "''${NIX_INDEX_DB}"' "" \
+      --replace nix-locate "${nix-index}/bin/nix-locate" \
+      --replace fzy "${fzy}/bin/fzy"
+  '';
+
+  nativeBuildInputs = [ makeWrapper ];
+  buildInputs = [ nix-index fzy ];
```
Comment (Artturin) [inline pkgs/tools/package-management/comma/default.nix:37]:
```suggestion
```

**Review by Artturin — CHANGES_REQUESTED** [review]

File: pkgs/tools/package-management/comma/default.nix (hunk 2 above)
Comment (Enzime) [reply pkgs/tools/package-management/comma/default.nix]: Fixed :+1:

File: pkgs/tools/package-management/comma/default.nix (hunk 3 above)
Comment (Enzime) [reply pkgs/tools/package-management/comma/default.nix]: Fixed :+1:

**Comment by siraben** [comment]
Result of `nixpkgs-review pr 150431` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>comma</li>
  </ul>
</details>

**Review by siraben — APPROVED** [review]

File: pkgs/tools/package-management/comma/default.nix [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  '';
+
+  installPhase = ''
+    install -Dm755 , -t $out/bin
+    ln -s $out/bin/, $out/bin/comma
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/nix-community/comma";
+    description = "Comma runs software without installing it";
```
Comment (siraben) [inline pkgs/tools/package-management/comma/default.nix:50]: `meta.description` shouldn't start with the package name.

```suggestion
    description = "Run software without installing it";
```

Comment (Enzime) [reply pkgs/tools/package-management/comma/default.nix]: Fixed

**Comment by 0x4A6F** [comment]
Thanks. Any objections for merging this? Should we wait for #153229?

**Review by siraben — APPROVED** [review]
Recommend merging since we have two major platforms supported.

**Pull request merged**